2-Draft-Accepted/RFCnnnn-PowerShell-as-Login-Shell.md

```diff
@@ -0,0 +1,79 @@
+---
+RFC: RFCnnnn
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 1.0
+Area: Console
+Comments Due: June 13, 2019
+Plan to implement: Yes
+---
+
+# Enable PowerShell as Login Shell on POSIX-based systems
+
+POSIX shells on Unix-based systems use a `-l` switch to start the shell as a login shell.
+This means the shell is not being started from an existing configured environment and
+needs to execute `/etc/profile` to populate environment variables along with other
+environment changes.
+
+## Motivation
+
+    As a PowerShell user,
+    I can use PowerShell as my login shell on Unix-based systems,
+    so that I can use PowerShell everywhere.
+
+## Specification
+
+Many existing tools expect `-l` to be accepted by a shell and to act as a login
+shell.
+
+`-l` (expanded form is `-LoadProfile`) will explicitly have pwsh load the PowerShell
```

Comment:
This seems strange to me. I can only imagine it will be confusing to people because `pwsh -LoadProfile -NoProfile` is possible (and it implies `pwsh` without it might not load the profile). But also just the conflation of a bash profile with the PowerShell profile is a can of worms.

It may as well be `-Login`. Or we could make it self-documenting: `-LoginShellSwitchCompatibilityNoOp`

Reply:
You can also do `pwsh -Interactive -NonInteractive`

Reply:
> It may as well be -Login.

:-) I wonder too - the RFC is "pwsh as __login__ shell" but suggested parameter is __LoadProfile__

Reply:
Based on the current proposal, `-l` will not perform the steps necessary to make it a login shell.  It could be named `-LoadPowerShellProfile`, but we don't call it `-NoPowerShellProfile`.  The use of it simply sets an internal property to load the powershell profile, so `-l -nop` would result in not loading the profile, while `-nop -l` would load the profile.

Reply:
> so -l -nop would result in not loading the profile, while -nop -l would load the profile

Wait, so it's not a no-op?

Also, having switches whose order matters doesn't mesh well with PowerShell's own semantics.

> You can already do pwsh -Interactive -NonInteractive

All the more reason to atone for past sins

Reply:
This is because pwsh.exe has to do it's own arg parsing and can't reply on PowerShell so it isn't as robust.  Although you could argue it's consistent with Unix conventions where last one wins (which enables Unix aliases to take switches that override the default defined in the alias...)